library/objective-c/EnvoyHTTPStreamImpl.m

```diff
@@ -179,9 +188,16 @@ - (instancetype)initWithHandle:(envoy_stream_t)handle
     return nil;
   }
 
-  _streamHandle = handle;
+  // Release the stream when the connection closes.
+  // It must be kept in memory while the stream is active in order to receive callbacks.
+  __weak typeof(self) weakSelf = self;
+  callbacks._onStreamClose = ^void() {
+    weakSelf._strongSelf = nil;
+  };
+
   // Retain platform callbacks
-  _platformCallbacks = callbacks;
+  self._platformCallbacks = callbacks;
```

Comment:
should we always use `self.`?

Reply:
Since these changed from ivars to properties, `self` is now required to use the generated getters from the compiler. I made `_strongSelf` a property so that it could be accessed within the closure above, then updated the rest for consistency. We can change the rest back if we want to